fastlane/lib/fastlane/actions/update_fastlane.rb

```diff
@@ -56,32 +57,49 @@ def self.run(options)
 
         update_needed.each do |tool_info|
           tool = tool_info[0]
+          gem_version = tool_info[1]
           local_version = Gem::Version.new(highest_versions[tool].version)
           update_url = FastlaneCore::UpdateChecker.generate_fetch_url(tool)
-          latest_version = FastlaneCore::UpdateChecker.fetch_latest(update_url)
-          UI.message("Updating #{tool} from #{local_version} to #{latest_version} ... 🚀")
+          latest_official_version = FastlaneCore::UpdateChecker.fetch_latest(update_url)
+
+          if options[:nightly]
+            UI.message("Updating #{tool} from #{local_version.to_s.yellow} to nightly build #{gem_version.to_s.yellow}... (last official release #{latest_official_version.to_s.yellow}) 🚀")
+          else
+            UI.message("Updating #{tool} from #{local_version.to_s.yellow} to #{latest_official_version.to_s.yellow}... 🚀")
+          end
 
           # Approximate_recommendation will create a string like "~> 0.10" from a version 0.10.0, e.g. one that is valid for versions >= 0.10 and <1.0
-          updater.update_gem tool, Gem::Requirement.new(local_version.approximate_recommendation)
+          updater.update_gem(tool, Gem::Requirement.new(gem_version))
```

Comment:
Hmm I looked through the source of `update_gem` and it seems like it always ignores the `updater.options` flags and instead looks to see if the passed `Gem::Requirement` contains any pre-releases. Could be that the updater doesn't know about the pre-releases, so maybe [`updater.fetch_remote_gems`](https://github.com/rubygems/rubygems/blob/master/lib/rubygems/commands/update_command.rb#L113) has to be called first?

Otherwise `updater.update_gem(tool, Gem::Requirement.new(options[:nightly] ? gem_version : local_version.approximate_recommendation))` could be used as a workaround?